### PR TITLE
DOC specgram() documentation now in numpy style

### PR DESCRIPTION
**Pull request by Exhor**
Changed documentation of lib/mlab.py for function specgram(), now in Numpy style.
@NelleV 

Thanks.

### CODE REVIEW

File: lib/matplotlib/mlab.py [hunk 1]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
-
-        specgram(x, NFFT=256, Fs=2,detrend=mlab.detrend_none,
-                window=mlab.window_hanning, noverlap=128,
-                cmap=None, xextent=None, pad_to=None, sides='default',
-                scale_by_freq=None, mode='default')
-
-    Compute and plot a spectrogram of data in *x*.  Data are split into
-    *NFFT* length segments and the spectrum of each section is
-    computed.  The windowing function *window* is applied to each
+    Compute and plot a spectrogram of data in x.  Data are split into
```
Comment (QuLogic) [inline lib/matplotlib/mlab.py]: Why remove the special formatting on the argument names? It makes them stand out more.

File: lib/matplotlib/mlab.py [hunk 2]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
+    specified with noverlap.
 
-      *x*: 1-D array or sequence
-        Array or sequence containing the data
+    Parameters
+    ----------
+    x : array_like
+        1-D array or sequence.
 
     %(Spectral)s
```
Comment (QuLogic) [inline lib/matplotlib/mlab.py]: This probably won't work correctly without fixing these interpolated portions as well.

**Comment by QuLogic** [comment]
It appears you have not set the config for your git client. All your commits are attributed to "Your Name <you@example.com>", which is probably not what you want.

File: lib/matplotlib/mlab.py (hunk 1 above)
Comment (NelleV) [inline lib/matplotlib/mlab.py]: It also makes it harder to read in plain text. I prefer it this way.

File: lib/matplotlib/mlab.py [hunk 3]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
-      *spectrum*: 2-D array
-        columns are the periodograms of successive segments
-
-      *freqs*: 1-D array
-        The frequencies corresponding to the rows in *spectrum*
+    noverlap : int, optional
+        The number of points of overlap between blocks.  The default 
+        value is 128.
+    mode : str, optional
+          What sort of spectrum to use, default is 'psd'.
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: There is trailing space at the end of this line. Can you please remove it?

File: lib/matplotlib/mlab.py [hunk 4]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
-      *freqs*: 1-D array
-        The frequencies corresponding to the rows in *spectrum*
+    noverlap : int, optional
+        The number of points of overlap between blocks.  The default 
+        value is 128.
+    mode : str, optional
+          What sort of spectrum to use, default is 'psd'.  
+          
+          'psd'
+              Returns the power spectral density.
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: There is trailing space at the end of this line. Can you please remove it?

File: lib/matplotlib/mlab.py [hunk 5]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
+              Returns the power spectral density.     
+              
+          'complex' 
+              Returns the complex-valued frequency spectrum.  
+              
+          'magnitude' 
+              Returns the magnitude spectrum.
+              
+          'angle' 
+              Returns the phase spectrum without unwrapping.
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: There is trailing space at the end of this line. Can you please remove it?

File: lib/matplotlib/mlab.py [hunk 6]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
-        The frequencies corresponding to the rows in *spectrum*
+    noverlap : int, optional
+        The number of points of overlap between blocks.  The default 
+        value is 128.
+    mode : str, optional
+          What sort of spectrum to use, default is 'psd'.  
+          
+          'psd'
+              Returns the power spectral density.     
+
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: There is whitespace in this line. Can you please remove it?

File: lib/matplotlib/mlab.py [hunk 7]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
+              
+          'complex' 
+              Returns the complex-valued frequency spectrum.  
+              
+          'magnitude' 
+              Returns the magnitude spectrum.
+              
+          'angle' 
+              Returns the phase spectrum without unwrapping.     
+
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: There is whitespace in this line. Can you please remove it?

File: lib/matplotlib/mlab.py [hunk 8]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
+          
+          'psd'
+              Returns the power spectral density.     
+              
+          'complex' 
+              Returns the complex-valued frequency spectrum.  
+              
+          'magnitude' 
+              Returns the magnitude spectrum.
+
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: There is whitespace in this line. Can you please remove it?

File: lib/matplotlib/mlab.py [hunk 9]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
+    magnitude_spectrum : A single spectrum, similar to having a single segment when mode is 'magnitude'.
+    angle_spectrum : A single spectrum, similar to having a single segment when mode is 'angle'.
+    phase_spectrum : A single spectrum, similar to having a single segment when mode is 'phase'.
 
-        :func:`phase_spectrum`
-            A single spectrum, similar to having a single segment when
-            *mode* is 'phase'.
+    Notes
+    -----
+    detrend and scale_by_freq only apply when *mode* is set to 'psd'.
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: Can you add a blank line at the end of the docstring?

File: lib/matplotlib/mlab.py [hunk 10]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
-
-    .. seealso::
-
-        :func:`psd`
-            :func:`psd` differs in the default overlap; in returning
-            the mean of the segment periodograms; and in not returning
-            times.
+    Returns
+    -------
+    spectrum: array_like
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: Can you add a space after spectrum and before the column? The rendering does not work when the space is missing.

File: lib/matplotlib/mlab.py [hunk 11]
```diff
@@ -1239,78 +1239,65 @@ def specgram(x, NFFT=None, Fs=None, detrend=None, window=None,
...
-            times.
+    Returns
+    -------
+    spectrum: array_like 
+        2-D array, columns are the periodograms of successive segments.
 
-        :func:`complex_spectrum`
-            A single spectrum, similar to having a single segment when
-            *mode* is 'complex'.
+    freqs: array_like
```
Comment (NelleV) [inline lib/matplotlib/mlab.py]: Can you add a space after spectrum and before the column? The rendering does not work when the space is missing.

**Comment by NelleV** [comment]
Great job on this PR! the documentation looks much better now.
I think if you fix the different problems I mentionned, we'll be good to go.

**Comment by NelleV** [comment]
It looks good and all the tests pass! Great job!

**Comment by tacaswell** [comment]
Does this render right with the replacement params?

**Comment by NelleV** [comment]
I'll check once I arrive home and add a screenshot to the PR.

**Comment by NelleV** [comment]
Here is how it renders.

specgram

**Comment by QuLogic** [comment]
The name on the first commit needs to be fixed. If you are not comfortable with rebasing before hand, we need to remember to do so before merging.

**Comment by Exhor** [comment]
Quite new to git. Rebase makes me scared.

**Comment by NelleV** [comment]
@Exhor I'll take care of it :)
Thanks for your contribution!

**Comment by NelleV** [comment]
Can you remind me your email address?

**Comment by Exhor** [comment]
t.corradi@gmail.com

**Comment by NelleV** [comment]
Thanks! I've rebased with this email address.

**Pull request closed without merging**